### PR TITLE
chore: loan feature deployment

### PR DESCRIPTION
**Pull request by lem-onade**
- [x] run on push to feat/loan
- [x] only publish/deploy apps and shared packages
- [x] alternate versioning for loan feature (along the lines of `1.1.0-loan.0+81e3b443`)

changed:
- checks.yml runs tests
- loan-canary.yml publishes canary releases for loan-related packages to alternate environment
- prod-release.yml publishes releases to production environments when PRs are closed to master (WIP, and unused in loan feature development)

### CODE REVIEW

**Comment by lem-onade** [comment]
@ddeboer I created a new loan-canary.yml workflow that creates new releases/images for the loan feature changes. These will be automatically published when PRs are closed to `feat/loan` branch.

I think we will need 4 new deployments for:

- solid-crs-manage (solid-crs-manage-loan?)
- solid-crs-presentation (solid-crs-presentation-loan?)
- solid-crs-semcom-node (solid-crs-semcom-node-loan?)
- solid-crs-semcom-components (solid-crs-components-loan?)

I'll wait for your feedback before making changes to the deployments config in loan-canary.yml.

**Comment by ddeboer** [comment]
Agreed, this requires new deployments. Do all of the four apps above change in the `feat/loan` branch?

We also need new URLs. What about https://solid-crs-loans.netwerkdigitaalerfgoed.nl?

**Comment by lem-onade** [comment]
> Agreed, this requires new deployments. Do all of the four apps above change in the `feat/loan` branch?

Only manage and presentation for now, but I expect changes in semcom-components, which would require a semcom-node as well. 

> We also need new URLs. What about https://solid-crs-loans.netwerkdigitaalerfgoed.nl?

Works for me!

**Comment by lem-onade** [comment]
Updated the deployment names

> All of these deployments use new Docker images, that should be built by this repository.

Would it also be alright if these images are named the same, but are tagged with `loans`? Can create new images for them as well.

**Comment by ddeboer** [comment]
> Would it also be alright if these images are named the same, but are tagged with loans? Can create new images for them as well.

Sure, if you prefer. Note that you’ll have to tag with the exact SHA too, so that would become `:loans-abc123`.

**Review by ddeboer — APPROVED** [review]

File: .github/workflows/loan-canary.yml [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+            NPM_TOKEN=${{secrets.DIGITA_PACKAGES}}
+
+      - name: Build and push semcom-components
+        uses: docker/build-push-action@v2
+        with:
+          context: packages/solid-crs-semcom-components/
+          push: true
+          tags: |
+            ${{ needs.variables.outputs.image_semcom_components }}:loans
+            ${{ needs.variables.outputs.image_semcom_components }}:${{ needs.publish-npm.outputs.version }}
```
Comment (ddeboer) [inline .github/workflows/loan-canary.yml:146]: Does this version include any `loan(s)` prefix?

Comment (lem-onade) [reply .github/workflows/loan-canary.yml:146]: I think this covers your previous [comment](https://github.com/netwerk-digitaal-erfgoed/solid-crs/pull/570#issuecomment-1056837911) as well?

Comment (ddeboer) [reply .github/workflows/loan-canary.yml:146]: Using tagged images in https://github.com/netwerk-digitaal-erfgoed/infrastructure/commit/59777e541f589ca134d819a753db5a296e3a7178, so this should be good to go now.

**Review by ddeboer — APPROVED** [review]

**Pull request merged**